rust/src/dataset/write.rs

```diff
@@ -115,8 +115,9 @@ pub async fn write_fragments(
     base_dir: &Path,
     schema: &Schema,
     data: SendableRecordBatchStream,
-    params: WriteParams,
+    mut params: WriteParams,
 ) -> Result<Vec<Fragment>> {
+    params.max_rows_per_group = std::cmp::min(params.max_rows_per_group, params.max_rows_per_file);
```

Comment:
This fix is needed from tests, but is already in #1092. Will rebase once that other is merged.